front/src/App.js

```diff
@@ -101,16 +101,31 @@ class App extends Component {
     this.setState({ connectionStatus: connectionTypes.CONNECTING });
   }
 
+  componentDidMount() {
+    this._isMounted = true;
+    this.connect();
+  }
+
   componentWillUnmount() {
+    this._isMounted = false;
     this.connection.close();
+    if (this.timer !== null) {
+      clearTimeout(this.timer);
+    }
   }
 
   onConnectionClose(event) {
-    this.setState({ connectionStatus: connectionTypes.DISCONNECTED });
+    if (this._isMounted)
+      this.setState({ connectionStatus: connectionTypes.DISCONNECTED });
+    this.timer = setTimeout(() => {
```

Comment:
Should the reconnect also happen is the component is not mounted?

Reply:
Indeed, I didn't notice it before